php-frontend/src/main/java/org/sonar/plugins/php/api/cfg/PhpCfgBranchingBlock.java

```diff
@@ -54,7 +56,8 @@ public Tree branchingTree() {
 
   @Override
   public Set<CfgBlock> successors() {
-    return SetUtils.immutableSetOf(trueSuccessor, falseSuccessor);
+    // `trueSuccessor` and `falseSuccessor` can be equal, but to comply with the API we need to have them in a Set
+    return Collections.unmodifiableSet(new HashSet<>(Arrays.asList(trueSuccessor, falseSuccessor)));
```

Comment:
```suggestion
    return Set.of(trueSuccessor, falseSuccessor);
```
I'm maybe missing something here :sweat_smile:

Reply:
Yes, I've added a comment above. Sometimes `trueSuccessor` and `falseSuccessor` may be equal, and then creation of a Set fails. However, a Set created from a List doesn't

Reply:
Ah! Thanks for pointing out! I forgot that `Set.of(...)` had such constraints, I thought that it would just ignore duplicate elements.